serve/graph/model/cursor.go

```diff
@@ -0,0 +1,101 @@
+package model
+
+import (
+	"encoding/base64"
+	"errors"
+	"strconv"
+	"strings"
+)
+
+// Cursor represents a string key of an element position in a sequential list of edges.
+type Cursor string
+
+// Create a cursor by base64-encoding the ID.
+func NewCursor(id string) Cursor {
+	return Cursor(base64.StdEncoding.EncodeToString([]byte(id)))
+}
+
+// UnmarshalGraphQL unmarshal incoming cursor into a local variable.
+func (c *Cursor) UnmarshalGraphQL(input interface{}) error {
+	var err error
+
+	switch input := input.(type) {
+	case string:
+		*c = Cursor(input)
+	case int32:
+		*c = Cursor(strconv.Itoa(int(input)))
+	default:
+		err = errors.New("wrong cursor type")
+	}
+
+	return err
+}
+
+// MarshalJSON encodes a cursor to JSON for transport.
+func (c Cursor) MarshalJSON() ([]byte, error) {
+	return strconv.AppendQuote(nil, string(c)), nil
+}
```

Comment:
why do we need a custom marshaller? Cursor is already a string and it will be serialized as intended.